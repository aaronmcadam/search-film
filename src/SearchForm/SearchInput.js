import styled from 'styled-components';

const SearchInput = styled.input`
  border-radius: 10px;
  font-size: 1.5rem;
  margin-right: 10px;
  padding: 10px;
`;

export default SearchInput;
