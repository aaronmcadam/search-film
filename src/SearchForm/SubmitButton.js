import styled from 'styled-components';

const SubmitButton = styled.button`
  font-size: 1.5rem;
  padding: 10px;
`;

export default SubmitButton;
