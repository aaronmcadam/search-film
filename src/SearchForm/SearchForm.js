import React from 'react';
import styled from 'styled-components';
import SearchInput from './SearchInput';
import SubmitButton from './SubmitButton';

const Wrapper = styled.form`
  margin-bottom: 2rem;
  padding-left: 1rem;
`;

function SearchForm(params) {
  return (
    <Wrapper data-role="search-form">
      <SearchInput
        data-role="search-input"
        type="text"
        placeholder="e.g. Blade Runner"
      />
      <SubmitButton>Search</SubmitButton>
    </Wrapper>
  );
}

export default SearchForm;
