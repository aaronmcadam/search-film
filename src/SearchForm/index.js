import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import SearchInput from './SearchInput';
import SubmitButton from './SubmitButton';

const Wrapper = styled.form`
  margin-bottom: 2rem;
  padding-left: 1rem;
`;

function SearchForm({ setInputRef, onSubmit }) {
  return (
    <Wrapper data-role="search-form" onSubmit={onSubmit}>
      <SearchInput
        data-role="search-input"
        type="text"
        placeholder="e.g. Blade Runner"
        innerRef={setInputRef}
      />
      <SubmitButton>Search</SubmitButton>
    </Wrapper>
  );
}

SearchForm.propTypes = {
  setInputRef: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

SearchForm.defaultProps = {
  setInputRef: () => {},
  onSubmit: () => {},
};

export default SearchForm;
