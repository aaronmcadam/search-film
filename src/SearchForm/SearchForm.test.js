import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

const setup = () => {
  const wrapper = shallow(<SearchForm />);

  return {
    wrapper,
  };
};

test('renders', () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});
