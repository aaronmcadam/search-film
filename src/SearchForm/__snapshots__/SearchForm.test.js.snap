// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.form
    data-role="search-form"
>
    <styled.input
        data-role="search-input"
        placeholder="e.g. Blade Runner"
        type="text"
    />
    <styled.button>
        Search
    </styled.button>
</styled.form>,
  "nodes": Array [
    <styled.form
      data-role="search-form"
>
      <styled.input
            data-role="search-input"
            placeholder="e.g. Blade Runner"
            type="text"
      />
      <styled.button>
            Search
      </styled.button>
</styled.form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <SearchForm />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.form
          data-role="search-form"
>
          <styled.input
                    data-role="search-input"
                    placeholder="e.g. Blade Runner"
                    type="text"
          />
          <styled.button>
                    Search
          </styled.button>
</styled.form>,
        "_debugID": 2,
        "_renderedOutput": <styled.form
          data-role="search-form"
>
          <styled.input
                    data-role="search-input"
                    placeholder="e.g. Blade Runner"
                    type="text"
          />
          <styled.button>
                    Search
          </styled.button>
</styled.form>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchForm />,
}
`;
